bundle/config/validate/schema_references.go

```diff
@@ -0,0 +1,130 @@
+package validate
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/databricks/cli/bundle"
+	"github.com/databricks/cli/libs/diag"
+	"github.com/databricks/cli/libs/dyn"
+)
+
+// Validate that any references to UC schemas defined in the DAB use the ${resources.schemas...}
+// syntax to capture the deploy time dependency.
+func SchemaReferences() bundle.ReadOnlyMutator {
+	return &schemaReferences{}
+}
+
+type schemaReferences struct{}
+
+func (v *schemaReferences) Name() string {
+	return "validate:schema_dependency"
+}
+
+func findSchemaInBundle(rb bundle.ReadOnlyBundle, catalogName, schemaName string) ([]dyn.Location, dyn.Path, bool) {
+	for k, s := range rb.Config().Resources.Schemas {
+		if s.CatalogName != catalogName || s.Name != schemaName {
+			continue
+		}
+		return rb.Config().GetLocations("resources.schemas." + k), dyn.NewPath(dyn.Key("resources"), dyn.Key("schemas"), dyn.Key(k)), true
+	}
+	return nil, nil, false
+}
+
+func (v *schemaReferences) Apply(ctx context.Context, rb bundle.ReadOnlyBundle) diag.Diagnostics {
+	diags := diag.Diagnostics{}
+	for k, p := range rb.Config().Resources.Pipelines {
+		// Skip if the pipeline uses hive metastore.
+		if p.Catalog == "" {
+			continue
+		}
+
+		schemaName := ""
+		fieldPath := dyn.Path{}
+		schemaLocation := []dyn.Location{}
+		switch {
+		case p.Schema == "" && p.Target == "":
+			diags = append(diags, diag.Diagnostic{
+				Severity: diag.Error,
+				Summary:  "Unity Catalog pipeline should have a schema or target defined",
+				Detail: `The target or schema field is required for UC pipelines. Reason: DLT
+requires specifying a target schema for UC pipelines. Please use the
+TEMPORARY keyword in the CREATE MATERIALIZED VIEW or CREATE STREAMING
+TABLE statement if you do not wish to publish your dataset.`,
+				Locations: rb.Config().GetLocations("resources.pipelines." + k),
+				Paths: []dyn.Path{
+					dyn.NewPath(dyn.Key("resources"), dyn.Key("pipelines"), dyn.Key(k), dyn.Key("schema")),
+					dyn.NewPath(dyn.Key("resources"), dyn.Key("pipelines"), dyn.Key(k), dyn.Key("target")),
+				},
+			})
+			continue
+		case p.Schema != "" && p.Target != "":
+			locations := rb.Config().GetLocations("resources.pipelines." + k + ".schema")
+			locations = append(locations, rb.Config().GetLocations("resources.pipelines."+k+".target")...)
+
+			diags = append(diags, diag.Diagnostic{
```

Comment:
Terraform already does not allow this today. Having it here just speeds up the validation and gives the user location information.

Reply:
This and other comments seems like valuable context when you're reading the code. Why not put it into code as comments?

Reply:
Thanks! Done.